### PR TITLE
Feat: color placeholders

### PR DESCRIPTION
**Pull request by andreaSimonePorceddu**
# Proposed changes

## Color Placeholders

Color placeholders allow to share a color between multiple keys with no need to duplicate it.

### Example 
```typescript
const colors: Colors = {
  // main palette
  dark: '#000000',
  light: '#ffffff',
  primary: '#06f',
  secondary: '#db2494',
  // functional palette
  background: '$light', // it will be resolved as #ffffff
  foreground: '$dark', // it will be resolved as #000000
  borderColor: '$primary', // it will be resolved as #06f
  ctaColor: '$secondary' // it will be resolved as #db2494
}
```

## Types of changes

- [ ] 🐛 Bugfix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality or enhancements)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] ✅ Tests (added tests for an existing feature)
- [x] 📚 Documentation Update (if none of the other choices apply)
- [ ] 🙌 Other (please, write a clear and concise description of the proposal in the section above)

## Checklist

- [x] 👀 I have read the [CONTRIBUTING](https://github.com/morfeojs/morfeo/blob/main/CONTRIBUTING.md) doc
- [x] ✅ Lint and unit tests pass locally with my changes
- [x] 🧪 I have added tests that prove my fix is effective or that my feature works
- [x] 📚 I have added the necessary documentation (if appropriate)
- [x] 🔀 Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/core/src/parsers/colors/colors.ts [hunk 1]
```diff
@@ -38,6 +40,14 @@ export function parseColor({
     }
   }
 
+  console.log(value);
```
Comment (mauroerta) [inline packages/core/src/parsers/colors/colors.ts:43]:
```suggestion
````

File: packages/core/src/parsers/colors/colors.ts [hunk 2]
```diff
@@ -38,6 +40,14 @@ export function parseColor({
...
+
+  const color = theme.getValue('colors', value as Color);
+
+  if (color && color.indexOf('$') === 0) {
+    return placeholderParser({ value, property, color });
```
Comment (mauroerta) [inline packages/core/src/parsers/colors/colors.ts:47]: Isn't it enough to call recursively the color parser:

```typescript
if (color && color.indexOf('$') === 0) {
   return parseColor({ value: color.substring(1), property });
}
```

Comment (andreaSimonePorceddu) [reply packages/core/src/parsers/colors/colors.ts:47]: in this way we are missing the condition that check if the original color is in place

Comment (mauroerta) [reply packages/core/src/parsers/colors/colors.ts:47]: right! nice catch!

File: packages/core/src/parsers/colors/placeholderParsers.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  if (!originalAssignedStyle) {
+    return {
+      [property]: value,
+    };
+  }
+
+  return baseParser({
+    value: originalValue as Color,
+    property: property === 'bg' ? 'backgroundColor' : property,
```
Comment (mauroerta) [inline packages/core/src/parsers/colors/placeholderParsers.ts:25]:
```suggestion
    property,
```

File: docs/docs/ThemeSpecification/colors.mdx [hunk 4]
```diff
@@ -152,6 +152,26 @@ an example of usage, and the correspondent output:
 | borderRightColor  | borderRightColor  | `{ borderRightColor: "secondary.light"} `  | `{"borderRightColor":"#e563b3"}`  |
 | borderBottomColor | borderBottomColor | `{ borderBottomColor: "secondary.light"} ` | `{"borderBottomColor":"#e563b3"}` |
 
+## Color Placeholders
+
+Color placeholders allow to share a color between multiple keys with no need to duplicate it.
```
Comment (mauroerta) [inline docs/docs/ThemeSpecification/colors.mdx:157]:
```suggestion
Color placeholders allow referring to another theme color by prefixing it with the character `$`:
```

**Review by mauroerta — APPROVED** [review]

**Pull request merged**